go/libraries/doltcore/sqle/dsess/autoincrement_tracker.go

```diff
@@ -138,7 +138,11 @@ func (a *AutoIncrementTracker) Next(ctx *sql.Context, tbl string, insertVal inte
 	}
 
 	if given >= curr {
-		a.sequences.Store(tbl, given+1)
+		nextVal := given
+		if a.canIncrementAutoIncVal(ctx, tbl, given) {
```

Comment:
just have this be `incrementAutoIncVal` that returns the nextVal and an ok boolean indicating if nextVal is valid.

Reply:
Sounds good 🫡